app/Http/Controllers/Service/Admin/DeliveriesController.php

```diff
@@ -122,6 +122,7 @@ public function store(AdminNewDeliveryRequest $request)
                 Voucher::select('id')
                     ->where('currentState', 'printed')
                     ->whereNull('delivery_id')
+                    ->where('code', 'like', $start['shortcode'] . '%') // Just vouchers that start with our shortcode
```

Comment:
While this is probably more efficient than using a regexp, it won't catch codes that are 
a) subsets of each other.
RVN0001
RVNT0001
b) different caps (as "like" in mysql is case insensetive)

Reply:
I'd suggest `where('code', 'REGEXP', "^{$start["shortcode"]}[0-9]+$")`
(though check that regexp)